presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchUtils.java

```diff
@@ -0,0 +1,118 @@
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.StandardTypes.ARRAY;
+import static com.facebook.presto.spi.type.StandardTypes.MAP;
+import static com.facebook.presto.spi.type.StandardTypes.ROW;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public final class ElasticsearchUtils
+{
+    private ElasticsearchUtils() {}
+
+    public static Block serializeObject(Type type, BlockBuilder builder, Object object)
+    {
+        if (ROW.equals(type.getTypeSignature().getBase())) {
+            return serializeStruct(type, builder, object);
+        }
+        if (MAP.equals(type.getTypeSignature().getBase()) || ARRAY.equals(type.getTypeSignature().getBase())) {
+            throw new IllegalArgumentException("Type not supported: " + type.getDisplayName());
+        }
+        serializePrimitive(type, builder, object);
+        return null;
+    }
+
+    private static Block serializeStruct(Type type, BlockBuilder builder, Object object)
+    {
+        if (object == null) {
+            requireNonNull(builder, "builder is null");
+            builder.appendNull();
+            return null;
+        }
+
+        boolean builderSynthesized = false;
+        if (builder == null) {
+            builderSynthesized = true;
+            builder = type.createBlockBuilder(null, 1);
+        }
+
+        BlockBuilder currentBuilder = builder.beginBlockEntry();
+        List<Type> typeParameters = type.getTypeParameters();
+
+        for (int i = 0; i < typeParameters.size(); i++) {
+            Optional<String> fieldName = type.getTypeSignature().getParameters().get(i).getNamedTypeSignature().getName();
+            String name = "";
+            if (fieldName.isPresent()) {
+                name = fieldName.get().toLowerCase(ENGLISH);
+            }
+            Object value = ((Map) object).get(name);
+            serializeObject(typeParameters.get(i), currentBuilder, value);
+        }
+
+        builder.closeEntry();
+        if (builderSynthesized) {
+            return (Block) type.getObject(builder, 0);
+        }
+        return null;
+    }
+
+    private static void serializePrimitive(Type type, BlockBuilder builder, Object object)
+    {
+        requireNonNull(builder, "builder is null");
+
+        if (object == null) {
+            builder.appendNull();
+            return;
+        }
+
+        if (type.equals(BOOLEAN)) {
+            type.writeBoolean(builder, (Boolean) object);
+        }
+        else if (type.equals(BIGINT)) {
+            type.writeLong(builder, Long.valueOf(String.valueOf(object)));
+        }
+        else if (type.equals(DOUBLE)) {
+            type.writeDouble(builder, (Double) object);
```

Comment:
I have two questions here:
- You cast to `Double` here, but you don't cast for long and int, you use `Long.valueOf`. Why don't you cast in those cases too?
- Why don't you use `Integer.valueOf` for the `INTEGER` case?

Reply:
my bad. will fix with Integer.valueOf and Double.valueOf
I'd prefer to use valueOf instead of casting. What do you think?